backup.sh

```diff
@@ -1,41 +1,108 @@
 #!/bin/sh -e
 
-if [ -z "${DBNAME}" ];
-then
+# Help function
+Help()
+{
+    echo "Backup database"
+    echo
+    echo "Usage: `basename ${0}` [ -D | --database ] [ -H | --host ] [ -U | -user ] [ -P | --password ] [ -B | --bucket ]"
+    echo "options:"
+    echo "-D | -d | ---database   Name of your database."
+    echo "-H | -h | ---host       Name of the host where the database is located. Must be either 'postgresql' or 'mariadb'"
+    echo "-U | -u | ---user       Name of the database user with enough access to dump the database."
+    echo "-P | -p | ---password   Password of the user with enough access to dump the database."
+    echo "-B | -b | ---bucket     Name of the S3 bucket where you want to dump your database."
+}
+while [[ $# -gt 0 ]];
+do case $1 in
+    -T | -t | --type) TYPE="$2"
+        shift 2
+        ;;
+    -D | -d | --database) DBNAME="$2"
+        shift 2
+        ;;
+    -H | -h | --host) DBHOST="$2"
+        shift 2
+        ;;
+    -U | -u | --user) DBUSER="$2"
+        shift 2
+        ;;
+    -B | -b | --bucket) BUCKET_DIR="$2"
+        shift 2
+        ;;
+    -P | -p | --password) DBPASSWORD="$2"
+        shift 2
+        ;;
+    --help)
+        Help
+        exit 0
+        ;;
+    *)
+        echo "Option '$1' is unknown. Type --help to display help message"
+        exit 1
+        ;;
+esac done
+
+# Check if all args are here
+if [ -z "$DBNAME" ]; then
   echo "ERROR: DBNAME env variable is not defined"
+  echo "Type --help to display help message"
   exit 1
 fi
 
-if [ -z "${DBHOST}" ];
-then
-  echo "ERROR: DBHOST env variable is not defined"
+if [ -z "$DBHOST" ]; then
+  echo "ERROR: DBHOST env variable must be either 'postgresql' or 'mariadb'"
+  echo "Type --help to display help message"
   exit 1
 fi
 
-if [ -z "${DBUSER}" ];
-then
+if [ -z "$DBUSER" ]; then
   echo "ERROR: DBUSER env variable is not defined"
+  echo "Type --help to display help message"
   exit 1
 fi
 
-if [ -z "${BUCKET_DIR}" ];
-then
-  echo "ERROR: BUCKET_DIR env variable is not defined"
+if [ -z "$PGPASSWORD" ]; then
```

Comment:
Should be `DBPASSWORD`

Reply:
I did the change to `DBPASSWORD` but I had an error saying postgres was looking a variable `PGPASSWORD` so I reverted.
Maybe it was before I fix the script, since then I haven't tested again with `DBPASSWORD`

Reply:
The error does not correspond to the check. Also DBPASSWORD is defined and never used:

```
$ grep DBPASSWORD backup.sh -C 1
        ;;
    -P | -p | --password) DBPASSWORD="$2"
        shift 2
--
if [ -z "$PGPASSWORD" ]; then
  echo "ERROR: DBPASSWORD env variable is not defined"
  echo "Type --help to display help message"
```